exchanges/kraken/kraken_wrapper.go

```diff
@@ -295,112 +320,128 @@ func (k *Kraken) Run() {
-	}
-
-	var products []string
-	for i := range pairs {
-		if strings.Contains(pairs[i].Altname, ".d") {
-			continue
+		format, err := k.GetPairFormat(assetType, false)
+		if err != nil {
+			return nil, err
 		}
-
-		base := assetTranslator.LookupAltname(pairs[i].Base)
-		if base == "" {
-			log.Warnf(log.ExchangeSys,
-				"%s unable to lookup altname for base currency %s",
-				k.Name,
-				pairs[i].Base)
-			continue
+		for i := range pairs {
+			if strings.Contains(pairs[i].Altname, ".d") {
+				continue
+			}
+			base := assetTranslator.LookupAltname(pairs[i].Base)
+			if base == "" {
+				log.Warnf(log.ExchangeSys,
+					"%s unable to lookup altname for base currency %s",
+					k.Name,
+					pairs[i].Base)
+				continue
+			}
+			quote := assetTranslator.LookupAltname(pairs[i].Quote)
+			if quote == "" {
+				log.Warnf(log.ExchangeSys,
+					"%s unable to lookup altname for quote currency %s",
+					k.Name,
+					pairs[i].Quote)
+				continue
+			}
+			products = append(products, base+format.Delimiter+quote)
 		}
-
-		quote := assetTranslator.LookupAltname(pairs[i].Quote)
-		if quote == "" {
-			log.Warnf(log.ExchangeSys,
-				"%s unable to lookup altname for quote currency %s",
-				k.Name,
-				pairs[i].Quote)
-			continue
+	case asset.Futures:
+		pairs, err := k.GetFuturesMarkets()
+		if err != nil {
+			return nil, err
+		}
+		for x := range pairs.Instruments {
+			if pairs.Instruments[x].Tradable {
+				products = append(products, pairs.Instruments[x].Symbol)
+			}
 		}
-		products = append(products, base+format.Delimiter+quote)
 	}
 	return products, nil
 }
 
 // UpdateTradablePairs updates the exchanges available pairs and stores
 // them in the exchanges config
 func (k *Kraken) UpdateTradablePairs(forceUpdate bool) error {
-	pairs, err := k.FetchTradablePairs(asset.UseDefault())
-	if err != nil {
-		return err
-	}
-
-	p, err := currency.NewPairsFromStrings(pairs)
-	if err != nil {
-		return err
+	assets := k.GetAssetTypes()
+	for x := range assets {
+		pairs, err := k.FetchTradablePairs(assets[x])
+		if err != nil {
+			return err
+		}
+		p, err := currency.NewPairsFromStrings(pairs)
+		if err != nil {
+			return err
+		}
+		err = k.UpdatePairs(p, assets[x], false, forceUpdate)
+		if err != nil {
+			return err
+		}
 	}
-	return k.UpdatePairs(p, asset.UseDefault(), false, forceUpdate)
+	return nil
 }
 
 // UpdateTicker updates and returns the ticker for a currency pair
 func (k *Kraken) UpdateTicker(p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
-	pairs, err := k.GetEnabledPairs(assetType)
-	if err != nil {
-		return nil, err
-	}
-	pairsCollated, err := k.FormatExchangeCurrencies(pairs, assetType)
-	if err != nil {
-		return nil, err
-	}
-	tickers, err := k.GetTickers(pairsCollated)
-	if err != nil {
-		return nil, err
-	}
-
-	for i := range pairs {
-		for c, t := range tickers {
-			pairFmt, err := k.FormatExchangeCurrency(pairs[i], assetType)
+	switch assetType {
+	case asset.Spot:
+		t, err := k.GetTicker(p)
```

Comment:
I should have been more explicit sorry. Please keep `tickerbatching: true,` and instead of using `GetTicker(p)` please use
`GetTickers()` like it used to be [here](https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/kraken/kraken_wrapper.go#L359)
Futures is ticker batching, lets keep spot ticker batching too